cranelift/codegen/src/machinst/compile.rs

```diff
@@ -49,6 +49,10 @@ pub fn compile<B: LowerBackend + TargetIsa>(
     log::debug!("Number of lowered vcode blocks: {}", vcode.num_blocks());
     trace!("vcode from lowering: \n{:?}", vcode);
 
+    // Update is_leaf based on actual VCode analysis
+    let is_leaf = !vcode.has_calls();
+    vcode.abi.set_is_leaf(is_leaf);
```

Comment:
Rather than use a mutable setter pattern with action-at-a-distance plumbing, let's compute the property directly from the ABI code. We already make a pass to find clobbers and so we should be able to detect this at the same time.